unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -169,13 +177,15 @@ public override void OnPoolGet()
             oldModel = new AvatarModel();
             model = new AvatarModel();
             lastAvatarPosition = null;
-            avatarName.SetName(String.Empty);
         }
 
         public override void Cleanup()
         {
             base.Cleanup();
 
+            if (playerStatus != null)
```

Comment:
You'd want to null `playerStatus` too. Remember `AvatarShape` is poolable. If `playerState` is not nulled, it will go inside the if at line 123 for recycled avatars.

Reply:
You just solved me a bug!